.github/workflows/servicenow.py

```diff
@@ -35,6 +35,45 @@ def find_string_in_json(json_data, target_string):
 
   return False
 
+def backoff_with_timeout(operation, max_retries=5, base_delay=1, max_delay=60, timeout=300):
+    start_time = time.time()
+    attempts = 0
+    while attempts <= max_retries and (time.time() - start_time) < timeout:
+        try:
+            return operation()  # Attempt the operation
+        except Exception as e:
+            attempts += 1
+            if attempts > max_retries or (time.time() - start_time) >= timeout:
+                raise  # Re-raise the exception if max retries or timeout is reached
+
+            delay = min(base_delay * (2 ** (attempts - 1)), max_delay) + random.uniform(0, 0.1 * base_delay)
+            time.sleep(delay)
+    raise TimeoutError("Operation timed out after {} seconds or {} retries, whatever came first.".format(timeout, max_retries))
```

Comment:
Should we add a print per try? So the action has better logs on re-trying

Reply:
I'll add the print in just before the operation() method execution.